File: src/main/resources/templates/js/domain/game/Status.js
export class Status {

    #dom
    #blackScore
    #whiteScore

    constructor() {
        this.#dom = this.#createDom()
    }

    setStatus(result) {
        this.#dom.innerHTML = `
            <p>black score : ${result.blackScore}</p><br>
            <p>white score : ${result.whiteScore}</p>
        `
    }

    getWinnerByScore() {
        if (this.#blackScore > this.#whiteScore) return 'black'
        if (this.#blackScore < this.#whiteScore) return 'white'
    }

    #createDom() {
        return document.createRange()
            .createContextualFragment(
                `<div id='status'></div>`
            ).firstElementChild
    }

    get dom() {
        return this.#dom
    }

}

File: src/main/java/domain/Status.java
package domain;

public class Status {

    private final double whiteScore;
    private final double blackScore;

    public Status(double whiteScore, double blackScore) {
        this.whiteScore = whiteScore;
        this.blackScore = blackScore;
    }

    public Player winner() {
        if (this.whiteScore > this.blackScore) {
            return Player.WHITE;
        }
        if (this.whiteScore < this.blackScore) {
            return Player.BLACK;
        }
        return null;
    }

    public double getWhiteScore() {
        return whiteScore;
    }

    public double getBlackScore() {
        return blackScore;
    }
}
